### ISSUE TITLE
Open links to other collective pages without popup

### ISSUE THREAD

**Issue by arnowelzel** [nextcloud/collectives#1227]
**Is your feature request related to a problem? Please describe.**
When I clink a link to another collective page in "view" mode, the page is not opened. Instead a popup opens with some information about this page and a "copy" icon. To open the page, you have to click the "info" box below. This is quite confusing as you don't really understand why the link does not just bring you to the linked page and what to do.

![image](https://github.com/nextcloud/collectives/assets/6613614/da1a22da-678c-4af9-b947-894d2bc92e6d)

**Describe the solution you'd like**
Leave links as they are and let the browser decide what to do. When clicking/tapping it, it will open the linked page, but people can also use their right mouse button or tap and hold it to open the context menu of the **browser** to choose what to do with it.

**Describe alternatives you've considered**
Make the behaviour configurable as global administrative setting.

In view mode show a popup when hovering the link, but still allow the link to be opened by just clicking it. In edit mode show the popup always when clicking the link, so positioning the cursor into the by clicking it, is still possible.

**Additional context**
Changing the behaviour of well known elements like links is bad for usability. People know what a "link" is: when a text is underlined and usually colored you can click or tap on it will open. This is how links work almost everywhere. There are also other well known patterns: clicking while holding [Ctrl] or [Cmd] will open the link in a new tab. Right-clicking it or tap and hold will open a context menu where you can decide what to do with the link - like copy it, open it in a new tab etc..

By inventing a new behaviour like "open a popup when you click a link" you force people to learn this new behaviour. It is at least confusing but may also be a real problem since the new popup even does not contain any clear indication where to click to open the linked page eventually.

One exception is when _editing_ documents, where you may want to be able to edit the link as well and not just open it. That's the reason why there are popups in _editors_ so one can choose what to do. But when _viewing_ documents, the behaviour of links should not be changed without a very good reason.

**Comment by arnowelzel** [comment]
Also see https://github.com/nextcloud/collectives/issues/1226

**Comment by mejo-** [comment]
Thanks for your report @arnowelzel.

> Leave links as they are and let the browser decide what to do. When clicking/tapping it, it will open the linked page, but people can also use their right mouse button or tap and hold it to open the context menu of the browser to choose what to do with it.

That's how it should be at the moment. You can still right-click on the link and select "Open link in new tab", also <kbd>Ctrl</kbd>-Click and (on Linux) middle click will open the link in a new tab straight away.

> One exception is when editing documents, where you may want to be able to edit the link as well and not just open it. 

Exactly that's the reason why we invented the link bubble (you were suggesting it yourself :wink:). And for consistency reasons we decided to make it behave the same way in edit and view mode. Otherwise people will be confused why clicking on a link behaves different in the different modes.

**Comment by arnowelzel** [comment]
> Thanks for your report @arnowelzel.
> 
> > Leave links as they are and let the browser decide what to do. When clicking/tapping it, it will open the linked page, but people can also use their right mouse button or tap and hold it to open the context menu of the browser to choose what to do with it.
> 
> That's how it should be at the moment. You can still right-click on the link and select "Open link in new tab", also Ctrl-Click and (on Linux) middle click will open the link in a new tab straight away.

On a mobile device it's no that easy to do "right-click". Not many people are aware of the context menu gesture of their mobile browsers and some browsers don't even support hat. And "right-click" is also not very self-explanatory for people who use a trackpad on a laptop or macOS devices.

> > One exception is when editing documents, where you may want to be able to edit the link as well and not just open it.
> 
> Exactly that's the reason why we invented the link bubble (you were suggesting it yourself 😉). And for consistency reasons we decided to make it behave the same way in edit and view mode. Otherwise people will be confused why clicking on a link behaves different in the different modes.

In https://github.com/nextcloud/collectives/issues/845 I suggested the link bubble for edit mode, not for view mode. For view mode (as in the PDF viewer) it makes no sense at all to open a bubble first when clicking the link. There is no need to do anything else besides letting the browser decide what to do with it. When people want just to copy the link instead of opening it, they can use the context menu of their device.

My preferred solution:

View mode: do not open the bubble at all, leave the link handling of the browser as it is.

Edit mode: make the link inside the bubble also clickable as link with `target="blank"` as attribute. This way nobody will accidentally close the current document one is working on, but you still can open the link easily.

Alternative:

If opening a bubble can not be avoided, make the link inside the bubble also clickable as link with `target="blank"` as attribute.

**Comment by sen4git** [comment]
I fully agree with @arnowelzel
> View mode: do not open the bubble at all, leave the link handling of the browser as it is.

Please - if possible - make it possible to keep it simple, as it was.

@mejo- Thanks for your work!

**Comment by ima-mes** [comment]
I also vote for making links act like _links_ in view mode. The current behavior is incredibly non-standard, and off-putting to people who are not Nextcloud-savvy users, just there to view content.

**Comment by derlucas** [comment]
I'm with others and would like to have the links in View mode act like regular Hyperlinks without Popup.
Would you Accept this Change as Pull Request if the behaviour is a configurable option?

**Comment by igefa-awelzel** [comment]
In addition - see how Confluence or Wikimedia handles links when viewing pages.

**Comment by arnowelzel** [comment]
In addition - see how Confluence or Wikimedia handles links when **viewing** pages. Both just open the link when clicking it - because this is how links work.

Maybe there is a misunderstanding here: it is about how links are handled when **viewing** pages.

**Comment by MarkWieczorek** [comment]
Every new user that I've talked to agrees with the above assessment of links in collectives: they are confusing. In fact, they are so confusing that we will probably stop using Collectives, even though it is an otherwise great app. 

One possible solution would be to allow the admin to configure how links should behave, either like normal links, or as popus.

**Comment by arnowelzel** [comment]
One more thing: yes, I am aware, that Nextcloud started inventing "rich links" for many apps like Deck or Talk. Probably the whole link component is just re-used everywhere and therefore can not be changed that easily. I also understand the motivation behind this to have a kind of preview before one opens a links.

But please consider changing this or make it at least configurable for Collectives. Even here in Github you can clearly see, that links just work as links when clicking them and there is not "rich preview" first.

**Comment by noantiq** [comment]
@arnowelzel I want to add to your mention of how Github handles links. When linking to a different issue for example, there is a rich preview - but that only appears on hover without obstructing the link itself or making it harder to just click it. I would be totally happy with rich previews in Collectives if they acted the same way.

**Comment by arnowelzel** [comment]
@noantiq Yes - I would prefer that too. Example in this issue here - you hover the link and a popup opens, but you can still just click the link itself and you don't have to find out where to click:

![Image](https://github.com/user-attachments/assets/6e37f950-283f-472a-a2d0-b6ee2e2bf176)

**Comment by arnowelzel** [comment]
Do you need help with that? Would pull requests which implement the behaviour similar to Github be accepted?

**Comment by arnowelzel** [comment]
Google Docs shows a popup in viewing and editing mode alike. So it looks like the implementation in Nextcloud was inspired by Google.

Editing mode:

<img width="1003" height="400" alt="Image" src="https://github.com/user-attachments/assets/86ec1999-377a-40f8-a35e-ce301ac953a8" />

Viewing mode:

<img width="949" height="340" alt="Image" src="https://github.com/user-attachments/assets/a700e9e4-a36e-4882-aba9-77d1562210b3" />

I am not sure, if Nextcloud should follow that. When it comes to Collectives, the reference is Confluence and not Google Docs. In Confluence the editing mode shows a popup as well, but there are more options ("Link bearbeiten" means "Edit link") what to do here including an icon to open the link in a new tab or options how to display the link:

<img width="1103" height="396" alt="Image" src="https://github.com/user-attachments/assets/43650706-4caa-4a5b-8c28-e65945e9ddd7" />

And in viewing mode Confluence does not show any popup at all - you click the link and it will just open, no matter if it is a link to another page inside Confluence or an external domain:

<img width="1239" height="527" alt="Image" src="https://github.com/user-attachments/assets/f94caeba-a440-4a65-84cd-6a13498b8188" />

And by the way - Confluence also allows to insert links to other pages in an easier way. You don't have to decide what kind of link you want - just type "/link" and a popup opens, where you can just enter what you want - if it is a page title, it will become a link to that page, but you also can just copy/paste a link to an external URL. There is no need to open a "Smart Picker" and the "Collectives Pages":

<img width="1039" height="1370" alt="Image" src="https://github.com/user-attachments/assets/55783dc3-08e8-4886-8132-dd98a4b8c3f7" />

<img width="1062" height="1034" alt="Image" src="https://github.com/user-attachments/assets/0517f02c-fe7d-44ab-81b5-458428644d9b" />

Maybe this can be addressed in the future as well. Linking to other Collectives pages should not be the exception which justifies the additional steps needed, but using `/link` in Collectives should directly bring you to the pages selection and not to a file selection box.

**Comment by mejo-** [comment]
I played a bit with adding a link open button at the end of each text link and so far I'm pretty happy with it:

Bright mode | dark mode
--- | ---
<img width="380" height="36" alt="Image" src="https://github.com/user-attachments/assets/961aef62-9b99-474b-8e2f-e4b07dd7176f" /> | <img width="380" height="36" alt="Image" src="https://github.com/user-attachments/assets/a5d46ccb-6ef0-4e93-b75a-d74014bb63f3" />

The button changes its color to `--color-primary-element` (the same as the link itself) on hover/focus and allows to open the link directly without having to open the link bubble first.

The idea would be to use the same click handler as we use in the link preview bubble for this button, i.e. open links to other collectives pages in the same tab and other links in a new tab. This would be additionally to <kbd>Ctrl</kbd>-click, which already allows to open a link without using the link bubble, but is not very discoverable.

What do you think about this solution @max-nextcloud @marcoambrosini @arnowelzel and others?

**Comment by max-nextcloud** [comment]
Sounds good to me. At first I was thinking about inverting the logic - the link opens on click and one gets the menu bubble with a button. But I guess the whole point is being able to place a cursor in the middle of the link by clicking there. So the opening functionality needs to be separate from the link.

**Comment by arnowelzel** [comment]
I agree that an additional button may help with that. But on the other hand this would now be even more confusing, that one click will open the link directly and another click may open the popup first. The current solution as in Nextcloud 33.0.4 and Collectives 4.4.0 is already nuch better then the situation when I first created this issue, since it now contains a button in the popup to open the link.

**Comment by marcoambrosini** [comment]
@mejo- I wouldn't add an inline "open link" button as it's not a common pattern. An arrow-out icon next to links is fine, but it should be decorative, not a separate click target.

From my pov, links should open directly on click in both view and edit mode. Many people work in Collectives with edit mode as their default and never explicitly switch to view mode, so the edit/view distinction matters less in practice. 

My suggestion: 
* On desktop: 
    * Always open links on click
    * Always show preview on hover 
* On mobile (where even less editing will happen): 
    * Always open links on tap
    * Require text selection and link-button press in the toolbar to edit links   

Reasoning:
* The link preview on click behavior penalizes the common case (navigating) to optimize for the rare one (editing the link itself). 
* The back button is always available, navigating to a new page is not a destructive action

**Comment by mejo-** [comment]
Thanks @marcoambrosini, I see your points, but the whole point of the link bubble was a design decision a few years back. The main point of the link bubble was that it's impossible to place the cursor in the link text in order to edit it when clicking on the link navigates away.

Google docs haves similar: clicking a link in both edit and view mode opens the link bubble:

<img width="535" height="235" alt="Image" src="https://github.com/user-attachments/assets/09cff6a3-4f46-42aa-aa68-2be6736add7e" />

**Comment by arnowelzel** [comment]
> Thanks [@marcoambrosini](https://github.com/marcoambrosini), I see your points, but the whole point of the link bubble was a design decision a few years back. The main point of the link bubble was that it's impossible to place the cursor in the link text in order to edit it when clicking on the link navigates away.
> 
> Google docs haves similar: clicking a link in both edit and view mode opens the link bubble:
> <img alt="Image" width="535" height="235" src="https://private-user-images.githubusercontent.com/3582805/590898846-09cff6a3-4f46-42aa-aa68-2be6736add7e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg1NzA1ODksIm5iZiI6MTc3ODU3MDI4OSwicGF0aCI6Ii8zNTgyODA1LzU5MDg5ODg0Ni0wOWNmZjZhMy00ZjQ2LTQyYWEtYWE2OC0yYmU2NzM2YWRkN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA1MTJUMDcxODA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzJkODY0MDEzYmExOGYwMTc1ZWJhNzVlMGFlNGJkYjMyZmM4YzQyOGY3YTUxMTNmZThhOTI0N2NjZGQ1OTBhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.hUBxfIDsX4eX-CinrLoFYxsAn3VlhGkgh5TR32o2jd4">

Yes, for EDIT mode showing a popup instead of opening the link is totally fine and expected - but not for READ mode.

In READ mode there is no cursor at all - just a link you can click. Having a popup opening before the link can be opened, is not the common pattern users are used to. Even if users want to copy that link - this is already possible with the context menu of every browser (right mouse button) out of the box, even mobile browsers (tap and hold the link until the context menu opens).

If you run Collectives with READ mode by default, it is confusing, when every link will open a popup first instead of just opening the linked page. Neither Mediawiki, DokuWiki nor Confluence behave like this when READING a page.

That's why I suggested not to have a popup for READ mode but only for WRITE mode. Please read my initial issue description - thank you!

Having an option to disable the popup for READ mode would be the best solution - people who want to keep that popup for READ mode can still have it, but others may just turn that off and only have it for WRITE mode.

**Comment by mejo-** [comment]
@arnowelzel we finally took the decision to change the link behaviour in Text (and that way also Collectives) to provide a better user experience. Thanks again for your feedback. I wasn't ready to change it back then and sorry it took me some time, but now your dream finally will come true 😉 https://github.com/nextcloud/text/pull/9202
